### PR TITLE
[BD-46] feat: algolia search

### PR DESCRIPTION
**Pull request by monteri**
Add Algolia search to docs site

Preview: https://deploy-preview-1246--paragon-openedx.netlify.app/

JIRA: https://openedx.atlassian.net/browse/PAR-795
Figma: https://www.figma.com/file/UWIyoulVBQ7AfGXCtG4y3q/DocSearch?node-id=0%3A1

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @monteri! I've created [BLENDED-1210](https://openedx.atlassian.net/browse/BLENDED-1210) to keep track of it in Jira.   More details are on the [BD-46](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3134685276/BD-46+Paragon+Blended+Support+and+Maintenance) project page.


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.

File: www/package.json [hunk 1]
```diff
@@ -3,6 +3,8 @@
   "description": "Paragon Pattern Library Documentation",
   "version": "1.0.0",
   "dependencies": {
+    "@docsearch/css": "^3.0.0",
+    "@docsearch/react": "^3.0.0",
```
Comment (adamstankiewicz) [inline www/package.json:7]: [curious/clarification] Hmm, it feels like the 27k line additions for package-lock.json file just by adding these 2 packages might be excessive? I am wondering if the package-lock.json file was possibly deleted and then re-created? For example, when I install these packages, I'm seeing much fewer line additions.

**Review by ihor-romaniuk — DISMISSED** [review]

File: www/package.json (hunk 1 above)
Comment (PKulkoRaccoonGang) [reply www/package.json]: This issue has been resolved, please take a look.

**Review by adamstankiewicz — APPROVED** [review]
Awesome! Tested it out locally and it's pretty slick. 🚀 🎉

**Pull request merged**